projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts

```diff
@@ -156,29 +156,57 @@ export function addPaymentMethod() {
 export function selectShippingAddress() {
   cy.server();
 
+  cy.route('GET', '/rest/v2/electronics-spa//users/current/carts/*').as(
```

Comment:
try carts/*deliverymode*/

it's different from the below one or something that gets from ###?fields=deliveryAddress(FULL),deliveryMode,paymentInfo(FULL)&lang=en&curr=USD

most likely at times we need a specific request because the wait might of gotten bypassed by another request which uses up our wait

Reply:
don't forget my comment below because I'm loss of ideas right now

Reply:
!!! I am literally  going through the docs again, and saw this https://docs.cypress.io/api/commands/route.html#Use-delays-for-responses

maybe this will help us (doing my-coupons) :(